iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -355,6 +355,16 @@ def sort_nodes(self):
         node_order = toposort_flatten(node_deps)
         self.nodes = [self.nodes[x] for x in node_order]
 
+        #Check inputs all in the beginning
+        root_node_section_passed = False
+        for node in self.nodes:
+            if not root_node_section_passed:
+                if 'input' not in str(node):
```

Comment:
There is a better way to do this check.  Checking the converted string is fragile because it relies on ancillary behavior that could change at any time.  For example it will break if we ever introduce a processing function that has the word input in its name.

The correct check is to iterate over the inputs of the node and check if `input[0].selector.input` is True.  You should do this check instead of anywhere where you are currently checking for the literal 'input' in a string